--- zimbra.sh	2022-05-15 15:16:08.354833731 -0500
+++ zimbra-sudo_runas_zimbra.sh	2022-05-15 14:26:00.000000000 -0500
@@ -33,16 +33,20 @@
 
    # append root pem so verifycrt can walk the chain
    cat "$_cfullchain" "$(dirname "$_cca")/../ISG_X1.pem" > "${_cca}.real"
-   /opt/zimbra/bin/zmcertmgr verifycrt comm "$_ckey" "$_ccert" "${_cca}.real" || return 1
+   sudo -E -u zimbra -n bash -c "/opt/zimbra/bin/zmcertmgr verifycrt comm "$_ckey" "$_ccert" "${_cca}.real" || return 1"
 
    #if it verifies we can deploy it
    $(which logger) -p local2.info NETWORK "Certificate has been Renewed for $_cdomain"
    cp -f "$_ckey" /opt/zimbra/ssl/zimbra/commercial/commercial.key
-   /opt/zimbra/bin/zmcertmgr deploycrt comm "$_ccert" "${_cca}.real" || return 1
+   sudo -E -u zimbra -n bash -c "/opt/zimbra/bin/zmcertmgr deploycrt comm "$_ccert" "${_cca}.real" || return 1"
    #/opt/zimbra/bin/ldap restart
    #/opt/zimbra/bin/zmmailboxdctl reload
    #/opt/zimbra/bin/zmproxyctl reload
    #/opt/zimbra/bin/zmmtactl reload
-   /opt/zimbra/bin/zmcontrol restart
+   sudo -E -H -u zimbra -n bash -c "/opt/zimbra/bin/zmcontrol restart"
+   sleep 2
+   sudo -E -H -u zimbra -n bash -c "/opt/zimbra/bin/zmloggerctl start"
+   sleep 2
+   sudo -E -H -u zimbra -n bash -c "/opt/zimbra/bin/zmstatctl start"
    return 0
 }
